app/models/mail_alias.rb

```diff
@@ -13,6 +13,7 @@ class MailAlias < ApplicationRecord
   before_validation :downcase_email
 
   before_save :set_smtp
+  after_commit :sync_mail_aliases
```

Comment:
Why is the mailgun function called on before_save and the improvmx one on after_commit?

Reply:
Because for the mailgun call you want to know if the smtp value has been changed (you only know that before save). The sync of the mail aliases can be done after the save since it doesn't need anything and is not needed to be done before save